File: AtlanticClojure/Grafcet.clj
( defstruct LocatedElement :location )
( defstruct NamedElement :name :location )
( defstruct Grafcet :name :location )
( defstruct Element :grafcet :name :location )
( defstruct Step :isInitial :isActive :action :grafcet :name :location )
( defstruct Transition :condition :grafcet :name :location )
( defstruct Connection :grafcet :name :location )
( defstruct StepToTransition :from :to :grafcet :name :location )
( defstruct TransitionToStep :from :to :grafcet :name :location )

File: AtlanticOsloM/Grafcet.m
module Grafcet {

	type LocatedElement {
		Location_Att : Text;
	};

	type NamedElement : LocatedElement {
		Name_Att : Text;
	};

	type Grafcet : NamedElement {
		Elements_ContRef : Element*;
		Connections_ContRef : Connection*;
	};

	type Element : NamedElement {
		Grafcet_Ref : Grafcet;
	};

	type Step : Element {
		IsInitial_Att : Logical;
		IsActive_Att : Logical;
		Action_Att : Text;
		IncomingConnections_Ref : TransitionToStep*;
		OutgoingConnections_Ref : StepToTransition*;
	};

	type Transition : Element {
		Condition_Att : Text;
		IncomingConnections_Ref : StepToTransition*;
		OutgoingConnections_Ref : TransitionToStep*;
	};

	type Connection : NamedElement {
		Grafcet_Ref : Grafcet;
	};

	type StepToTransition : Connection {
		From_Ref : Step;
		To_Ref : Transition;
	};

	type TransitionToStep : Connection {
		From_Ref : Transition;
		To_Ref : Step;
	};
}